pkgs/by-name/co/containerd/package.nix

```
@@ -1,56 +1,85 @@
-{ lib
-, fetchFromGitHub
-, buildGoModule
-, btrfs-progs
-, go-md2man
-, installShellFiles
-, util-linux
-, nixosTests
-, kubernetes
+{
+  lib,
+  stdenv,
+  btrfs-progs,
+  fetchFromGitHub,
+  go,
+  go-md2man,
+  kubernetes,
+  nix-update-script,
+  nixosTests,
+  util-linux,
+  btrfsSupport ? btrfs-progs != null,
 }:
 
-buildGoModule rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "containerd";
-  version = "1.7.23";
+  version = "2.0.0";
+
+  outputs = [
+    "out"
+    "doc"
+    "man"
+  ];
 
   src = fetchFromGitHub {
     owner = "containerd";
     repo = "containerd";
-    rev = "v${version}";
-    hash = "sha256-vuOefU1cZr1pKCYHKyDBx/ohghgPlXhK3a38PQKH0pc=";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-DFAP+zjBYP2SpyD8KXGvI3i/PUZ6d4jdzGyFfr1lzj4=";
   };
 
-  vendorHash = null;
+  postPatch = "patchShebangs .";
+
+  strictDeps = true;
 
-  nativeBuildInputs = [ go-md2man installShellFiles util-linux ];
+  nativeBuildInputs = [
+    go
+    go-md2man
+    util-linux
+  ];
 
   buildInputs = [ btrfs-progs ];
```

```suggestion
  buildInputs = lib.optional (!btrfsSupport) btrfs-progs;
```

This will exclude `btrfs-progs` from the build closure when user explicitly sets `btrfsSupport = false`

---

Or maybe just like that, if you're supposed to disable btrfs using `override { btrfs-progs = null; }`

```patch
-  tags = lib.optional (!btrfsSupport) "no_btrfs";
+  tags = lib.optional (btrfs-progs != null) "no_btrfs";
```